RingDelayBufferTest: refactor includes and span creation

davidchocholaty opened the pull request:
The PR just improves a little bit the `RingDelayBufferTest`. The unused include is removed and the multiple span creation from one `SampleBuffer` is avoided.

davidchocholaty committed:
RingDelayBufferTest: remove unused include

The commit removes unused include of the "util/sample.h" based
on the previous changes.

davidchocholaty committed:
RingDelayBufferTest: avoid multiple span creation

This commit avoids multiple span creation from the one sample buffer
(exactly SampleBuffer). Instead of that, the span is created once
and stored in the variable.

Swiftb0y reviewed (commented):
while you're at it, you can try to eliminate all the superfluous `mixxx::spanutil::spanFromPtrLen(..., numSamples)` calls as well.

Swiftb0y commented:
though fyi, optimizing for calls to `std::span` is not really worth much effort. at least not when the goal is performance. std::span is a very, very cheap type to construct.

davidchocholaty committed:
RingDelayBufferTest: eliminate spanFromPtrLen

This commit avoids multiple span creation from the input buffer.
Instead of that, the span is created once and stored in the variable.

davidchocholaty commented:
> though fyi, optimizing for calls to `std::span` is not really worth much effort. at least not when the goal is performance. std::span is a very, very cheap type to construct.

Okay, thank you for informing me. It just didn't make much sense to me to duplicate the same operation multiple times (but TBH I haven't known how span is a cheap operation).

davidchocholaty commented:
> while you're at it, you can try to eliminate all the superfluous `mixxx::spanutil::spanFromPtrLen(..., numSamples)` calls as well.

All multiple used spans should be deduplicated now. Although I kept the just once used span as were because it didn't make much sense to me to store them in a variable.

Swiftb0y commented on src/test/ringdelaybuffer_test.cpp:
```diff
@@ -49,35 +48,38 @@ TEST_F(RingDelayBufferTest, ReadWriteNoDelayTest) {
     mixxx::SampleBuffer output(numSamples);
 
+    std::span<CSAMPLE> outputSpan = output.span();
+    std::span<CSAMPLE> inputBufferSpan = mixxx::spanutil::spanFromPtrLen(inputBuffer, numSamples);
```
Build fails because `spanFromPtrLen` can only create a `std::span<const CSAMPLE>` since the buffer it observes is also `const`. Either of the below suggestions would work. I prefer the `auto` version since the type is clear and the line is already somewhat long. 

```suggestion
    std::span<const CSAMPLE> inputBufferSpan = mixxx::spanutil::spanFromPtrLen(inputBuffer, numSamples);
```

```suggestion
    auto inputBufferSpan = mixxx::spanutil::spanFromPtrLen(inputBuffer, numSamples);
```

davidchocholaty replied:
Auto looks better in this situation to me too. Thank you.

Swiftb0y reviewed (approved):
lgtm. waiting for ci. thanks
